.github/workflows/ci-cd.yml

```diff
@@ -0,0 +1,65 @@
+name: CI/CD Workflow
+
+on:
+  push:
+    branches:
+      - 'main'
+      - 'feature/*'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '20'
+
+      - name: Install dependencies using pnpm
+        run: |
+          corepack enable
+          pnpm install
+
+      - name: Run tests
+        run: pnpm test
+
+  deploy:
+    if: github.ref == 'refs/heads/main'
+    needs: build
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '20'
+
+      - name: Install dependencies using pnpm
+        run: |
+          corepack enable
+          pnpm install
+
+      - name: Build project
+        run: pnpm run build
+
+      - name: Log in to Docker Hub
+        run: |
+          echo "${{ secrets.DOCKER_PASSWORD }}" | docker login -u "${{ vars.DOCKER_USERNAME }}" --password-stdin
+
+      - name: Build Docker image
+        run: |
+          TAG="latest"  # You can adjust the tag (e.g., based on version, commit, etc.)
+          docker build -t ${{ vars.DOCKER_USERNAME }}/my-app:$TAG .
+          docker tag ${{ vars.DOCKER_USERNAME }}/my-app:$TAG ${{ vars.DOCKER_USERNAME }}/my-app:latest
+
+      - name: Push Docker image to Docker Hub
+        run: |
+          docker push ${{ vars.DOCKER_USERNAME }}/my-app:$TAG
+          docker push ${{ vars.DOCKER_USERNAME }}/my-app:latest
```

Comment:
I don't know how it is currently named (so just a guess from my side), but the target name of the image is probably not going to be `my-app` ?

Reply:
yep we doing monorepo so there is actually do different docker images for now (soon to be more)

https://hub.docker.com/r/softfl0w/client
https://hub.docker.com/r/softfl0w/landing

I think for packaging we could reuse the old pipeline?


https://github.com/softflow24/data-river/blob/25b7cc92041c18008719198ab739e462b17a293f/.github/workflows/develop-build.yml#L28-L68

Reply:
Referred `client` as the name from here

<img width="500" alt="Screenshot 2024-10-13 at 10 40 22 PM" src="https://github.com/user-attachments/assets/5766276e-d2c7-42b7-9d30-9212d3edf900">